packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ModelController.java

```diff
@@ -295,9 +299,37 @@ ResponseEntity<List<ModelConfiguration>> getModelConfigurationsForModelId(
 						log.error("Unable to get the document " + documentId, e);
 					}
 				});
+
+				// Find the Dataset Assets linked via provenance to the model configuration
+				final ProvenanceQueryParam datasetSearchBody = new ProvenanceQueryParam();
+				datasetSearchBody.setRootId(config.getId());
+				datasetSearchBody.setRootType(ProvenanceType.MODEL_CONFIGURATION);
+				datasetSearchBody.setTypes(List.of(ProvenanceType.DATASET));
+				final Set<String> datasetIds = provenanceSearchService.modelConfigFromDataset(datasetSearchBody);
```

Comment:
This method was written to throw a `IllegalArgumentException` in some cases - if thats going to be an expected path then we should throw some/all of this in a try/catch to handle that failure case... but if its a super rarety and thats just a bit of safety code its probably fine!

Reply:
Good idea, I've make the catch for all exceptions now!